### ISSUE TITLE
Bug: css/no-important false positives when `!important` required

### ISSUE THREAD

**Issue by adamlui** [eslint/css#162]
### Environment

ESLint version: v9.27.0
@eslint/css version: v1.10.34
Node version: v22.15.0
npm version: v11.3.0
Operating System: win32 10.0.19045

### Which language are you using?

stylesheet

### What did you do?

<details>
<summary>Configuration</summary>

```js
import js from '@eslint/js'
import globals from 'globals'
import css from '@eslint/css'
import html from '@html-eslint/eslint-plugin'
import htmlParser from '@html-eslint/parser'
import importPlugin from 'eslint-plugin-import'
import json from '@eslint/json'
import markdown from '@eslint/markdown'
import regexp from 'eslint-plugin-regexp'
import stylisticJS from '@stylistic/eslint-plugin'
import yml from 'eslint-plugin-yml'

export default [
    { ignores: ['**/*.min.js', '**/package-lock.json', '**/rising-particles/'] },
    {
        files: ['**/*.{js,mjs}'],
        languageOptions: {
            ecmaVersion: 'latest', sourceType: 'script',
            globals: {
                ...globals.browser, ...globals.greasemonkey, ...globals.node, api: 'readonly', apis: 'writable',
                app: 'writable', browserAPI: 'readonly', buttons: 'writable', chatbar: 'writable', chatgpt: 'readonly',
                checkBtnsToClick: 'readonly', chrome: 'readonly', config: 'writable', CryptoJS: 'readonly',
                deepseek: 'readonly', dom: 'readonly', env: 'writable', feedback: 'readonly',
                fontSizeSlider: 'writable', get: 'readonly', GM_cookie: 'readonly', hljs: 'readonly', icons: 'writable',
                importScripts: 'readonly', infinity: 'readonly', inputEvents: 'readonly', JSON5: 'readonly',
                ipv4: 'readonly', log: 'readonly', marked: 'readonly', menus: 'writable', modals: 'writable',
                notify: 'readonly', prompts: 'readonly', renderMathInElement: 'readonly', replyBubble: 'writable',
                session: 'readonly', settings: 'writable', show: 'readonly', sites: 'writable', styles: 'readonly',
                sync: 'readonly', syncConfigToUI: 'readonly', toggle: 'readonly', toggleMode: 'readonly',
                toggles: 'writable', toolbarMenu: 'writable', tooltip: 'writable', ui: 'readonly', update: 'readonly',
                updateCheck: 'readonly', userscript: 'readonly', xhr: 'writable'
            }
        },
        plugins: { 'import': importPlugin, 'js-styles': stylisticJS, regexp },
        rules: {
            ...js.configs.recommended.rules,
            ...importPlugin.flatConfigs.recommended.rules,
            ...regexp.configs['flat/recommended'].rules,
            'indent': 'off', 'no-unexpected-multiline': 'off', 'key-spacing': 'off', // allow whitespace anywhere
            'js-styles/no-trailing-spaces': 'error', // ...except at ends of lines
            'js-styles/max-len': ['error', { 'code': 120, // limit lines to 120 chars except if containing...
                'ignoreComments': true, 'ignoreStrings': true, // ...trailing/own-line comments, quoted strings...
                'ignoreTemplateLiterals': true, 'ignoreRegExpLiterals': true }], // ...or template/regex literals
            'js-styles/no-extra-semi': 'error', // disallow unnecessary semicolons
            'quotes': ['error', 'single', // enforce single quotes...
                { 'allowTemplateLiterals': true }], // ...except backticks to avoid escaping quotes
            'comma-dangle': ['error', 'never'], // enforce no trailing commas in arrays or objects
            'no-constant-condition': 'off', // allow constant conditions
            'no-empty': 'off', // allow empty blocks
            'no-inner-declarations': 'off', // allow function declarations anywhere
            'no-useless-escape': 'off', // allow all escape chars cause ESLint sucks at detecting truly useless ones
            'no-unused-vars': ['error', { 'caughtErrors': 'none' }] // allow unused named args in catch blocks
        }
    },
    { files: ['**/chatgpt.js'], languageOptions: { globals: { chatgpt: 'off' }}},
    { files: ['**/deepseek.js'], languageOptions: { globals: { deepseek: 'off' }}},
    { files: ['**/*.css'], language: 'css/css', ...css.configs.recommended },
    {
        files: ['**/*.html'], languageOptions: { parser: htmlParser }, plugins: { '@html-eslint': html },
        rules: {
            ...html.configs['flat/recommended'].rules,
            '@html-eslint/require-title': 'off', // allow missing title tags
            '@html-eslint/quotes': 'off', // allow unquoted attrs for compactness
            '@html-eslint/attrs-newline': 'off', // allow multi-attrs in single line
            '@html-eslint/require-img-alt': 'off', // allow img tags w/o alt attributes
            '@html-eslint/indent': 'off', '@html-eslint/element-newline': 'off' // allow whitespace anywhere
        }
    },
    { files: ['**/*.json'], language: 'json/json', ...json.configs.recommended },
    { files: ['**/*.json5'], language: 'json/json5', ...json.configs.recommended },
    {
        files: ['**/*.md'], language: 'markdown/commonmark', plugins: { markdown },
        rules: {
            ...markdown.configs.recommended[0].rules,
            'markdown/heading-increment': 'off', // allow headings to skip levels
            'markdown/fenced-code-language': 'off', // allow code blocks w/ no language specified
            'markdown/no-missing-label-refs': 'off' // allow missing label references
        }
    },
    { files: ['**/*.mjs'], languageOptions: { sourceType: 'module' }},
    { files: ['**/*.{yaml,yml}'], ...yml.configs['flat/standard'][1] }
]
```
</details>

```css
/* General */
html { height: fit-content ; min-height: 89px }
body {
    width: 275px ; margin: 0 ; font-size: .905rem ;
    font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto,
        "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "Helvetica Neue", sans-serif    
}

/* Color/fade mods */
.highlight-on-hover:hover { background: rgb(100,149,237) }
.highlight-on-hover:hover span:not(.track), .highlight-on-hover:hover svg, span.menu-icon.highlight-on-hover:hover img {
    filter: invert(1) } /* invert setting labels on hover */
.disabled { opacity: 0.3 !important; pointer-events: none }

/* Loader */
.loading-bg {
    background-color: white ; width: 100% ; height: 100% ; position: absolute ; z-index: 1111 ;
    display: inline-grid ; align-content: center ; justify-content: center /* center spinner */
}
.loading-spinner {
    width: 19px ; aspect-ratio: 1 ; border-radius: 50% ; border: 3px solid #000 ;
    animation: loader-move-head-tail 0.8s infinite linear alternate, loader-rotate 1.6s infinite linear
}
@keyframes loader-move-head-tail {
     0% { clip-path: polygon(50% 50%, 0 0, 50% 0, 50% 0, 50% 0, 50% 0, 50% 0) }
     12.5% { clip-path: polygon(50% 50%, 0 0, 50% 0, 100% 0, 100% 0, 100% 0, 100% 0) }
     25% { clip-path: polygon(50% 50%, 0 0, 50% 0, 100% 0, 100% 100%, 100% 100%, 100% 100%) }
     50% { clip-path: polygon(50% 50%, 0 0, 50% 0, 100% 0, 100% 100%, 50% 100%, 0 100%) }
     62.5% { clip-path: polygon(50% 50%, 100% 0, 100% 0%, 100% 0, 100% 100%, 50% 100%, 0 100%) }
     75% { clip-path: polygon(50% 50%, 100% 100%, 100% 100%, 100% 100%, 100% 100%, 50% 100%, 0 100%) }
     100% { clip-path: polygon(50% 50%, 50% 100%, 50% 100%, 50% 100%, 50% 100%, 50% 100%, 0 100%) }
}
@keyframes loader-rotate { 
    0% { transform: scaleY(1) rotate(0deg) } 49.99% { transform: scaleY(1) rotate(135deg) }
    50% { transform: scaleY(-1) rotate(0deg) } 100% { transform: scaleY(-1) rotate(-135deg) }
}

/* Header */
header {
    border-bottom: solid 1px lightgrey ; padding: 5px 5px 5px 0 ; margin: 0 ;
    height: 38px ; display: flex; background: white ; align-items: center }
.logo { margin: 0 8px 0 12px ; position: relative ; top: 3px }
.menu-title { font-size: 0.85rem ; font-weight: 600 ; padding-right: 3px }
.master-toggle { margin-left: auto ; display: flex }

/* Menu item elements */
.menu-entry {
    position: relative ; align-items: center ; border-bottom: 1px solid lightgrey ;
    display: flex ; min-height: 2rem ; padding: 0 14px 0 2px ; white-space: nowrap ; font-size: 91%
}
.menu-icon { padding: 8px ; justify-content: center }
.menu-entry > label > .track { transform: scale(0.95) ; top: 1px } /* make child toggles smaller */
.menu-right-elem { display: flex ; height: 33px ; align-items: center ; padding: 0 11.5px ; margin-right: -14px }
.menu-caret { position: absolute ; right: 14px ; transform: rotate(-90deg) }
.menu-right-elem, .menu-caret { filter: drop-shadow(5px 5px 5px rgba(0, 0, 0, 0.1)) }
.categorized-entries { /* add left-stripe */
    border-left: 5px solid transparent ; height: 0 ;
    overflow: hidden ; /* for y-pos calc */
    border-image: linear-gradient(transparent, rgb(161 161 161)) 30 100%
}
.anchored { pointer-events: none } /* disable collapse */
.expanded > span { font-weight: 600 ; transform: scale(1.05) ; margin-left: 5px }

/* Slider */
div.menu-entry:has(> input.slider) { flex-wrap: wrap ; padding-top: 2px }
div.menu-entry > input.slider {
  --track-filled-color: #000 ; --track-empty-color: #E0E0E0 ; --thumb-color: #000 ; --thumb-border: #fff ;
  --track-fill-percent: 100% ; padding: 7px 0 ; width: 100% ; margin: -4px -5px 7px 8px ; cursor: pointer ;
   -webkit-appearance: none ; appearance: none
}
.slider::-webkit-slider-runnable-track {
    height: 6px ; border-radius: 10px ;
    background: linear-gradient(to right,
        var(--track-filled-color) 0%, var(--track-filled-color) var(--track-fill-percent),
        var(--track-empty-color) var(--track-fill-percent), var(--track-empty-color) 100%
    )
}
.slider::-moz-range-track {
    height: 6px ; border-radius: 10px ;
    background: linear-gradient(to right,
        var(--track-filled-color) 0%, var(--track-filled-color) var(--track-fill-percent),
        var(--track-empty-color) var(--track-fill-percent), var(--track-empty-color) 100%
    )
}
.slider::-webkit-slider-thumb {
   -webkit-appearance: none ; width: 13px ; height: 27.5px ; background: var(--thumb-color) ; margin-top: -11px ;
    border: 4px solid var(--thumb-border) ; border-radius: 16px ; cursor: ew-resize ; transition: transform 0.05s ease
}
.slider::-moz-range-thumb {
    width: 5.5px ; height: 20px ; background: var(--thumb-color) ; margin-top: -11px ;
    border: 4px solid var(--thumb-border) ; border-radius: 16px ; cursor: ew-resize ; transition: transform 0.05s ease
}
.slider::-webkit-slider-thumb:hover { transform: scaleX(1.325) }
.slider::-moz-range-thumb:hover { transform: scaleX(1.325) }

/* About entry */
div#about > div.menu-icon { padding: 8px 10px 8px 11px }
.ticker {
    align-content: center ; overflow: hidden ; font-size: 10px ; width: 105px ;
  --mask: linear-gradient(to right, transparent, black 20%, black 89%, transparent) ;
    mask-image: var(--mask) ; -webkit-mask-image: var(--mask) /* eslint-disable-line */
}
.ticker-em { color: green } .highlight-on-hover:hover .ticker-em { color: #28ee28 }
.ticker > div { animation: ticker linear 60s infinite } 
@keyframes ticker { 0% { transform: translateX(100%) } 100% { transform: translateX(-2000%) }}

/* Toggle elements */
.toggle .track {
    display: block ; border-radius: 7px ; position: relative ; transform: scale(0.95) ; background: white ; 
    border: 1px solid black ; top: 1px ; height: 10px ; width: 18px ; transition: background 0.08s ease-in-out
}
.toggle.on .track { background: black ; transition: background 0.15s ease-in-out }
.toggle .track::before { /* knob */
    content: "" ; position: absolute ; height: 0.625rem ; width: 0.625rem ; left: -0.04rem ; top: -0.05rem ;
    border: 1px solid black ; border-radius: 50% ; background: white ; transition: transform 0.08s ease-in-out
}
.toggle.on .track::before { transform: translateX(9px) ; transition: transform 0.15s ease-in-out }
  
/* Footer */
footer { display: flex ; align-items: center ; background: #f5f5f5 ; height: 40px ; padding: 0 7px }
.chatgptjs-logo { opacity: 0.5 } .chatgptjs-logo:hover { opacity: 1 ; transition: 0.25s }
footer > .menu-icon {
    display: flex ; opacity: 0.7 ; align-items: center ; padding: 8px 5px ;
    box-sizing: border-box ; height: 96% ; width: 26px
}
footer img[src*=question-mark] { position: relative ; top: 1px }

/* Non-baseline features */
@supports (cursor: pointer) { .highlight-on-hover:hover, .toggle .track, .chatgptjs-logo { cursor: pointer }}
@supports (overflow: clip) { body { overflow: clip }}
@supports selector(:has(> .chatgptjs-logo)) {
      footer > span:has(> .chatgptjs-logo) { display: flex ; align-items: center ; flex-grow: 1 ; padding-left: 2px }}
@supports (user-select: none) { body, button, input, select, textarea { user-select: none }}
```

### What did you expect to happen?

No errorr the line

```css
.disabled { opacity: 0.3 !important; pointer-events: none }
```

### What actually happened?

ESLint errors `Unexpected !important flag found`

![Image](https://github.com/user-attachments/assets/b22e858e-c7f8-457c-8f66-ce0f26ac11f7)

### Link to Minimal Reproducible Example

https://github.com/adamlui/ai-web-extensions/actions/runs/15365546827/job/43237973703

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Without the `!important` the `opacity` doesn't take effect for example if you toggle Full-Window from popup menu on chatgpt.com
